Notification controller & security

oleksandrYeriemieiev opened the pull request:
What’s included
* Added `NotificationController` — initial implementation for issues #179, #181,#182, #183
   `GET /notifications` – returns the user’s notifications  
   `POST /notifications/{id}/markAsRead`  
   `POST /notifications/markAllAsRead`  
   `POST /notifications/markAsReadBulk`
**Draft version**:only the basic.

oleksandrYeriemieiev committed:
feat: NotificationController draft  for review

natri0 commented on service/src/main/java/greencity/controller/NotificationController.java:
```diff
@@ -0,0 +1,69 @@
+    private final NotificationService notificationService;
+    private final UserService userService;
+
+    /** Отримати всі нотифікації поточного користувача. */
```
comments are usually in english
```suggestion
    /** Retrieve all notifications for the current user. */
```

natri0 commented on service/src/main/java/greencity/controller/NotificationController.java:
```diff
@@ -0,0 +1,69 @@
+        return ResponseEntity.ok(notificationService.getAllNotificationsForUser(userId));
+    }
+
+    /** одну нотифікацію як прочитану. */
```
```suggestion
    /** Mark one notification as read. */
```

natri0 commented on service/src/main/java/greencity/controller/NotificationController.java:
```diff
@@ -0,0 +1,69 @@
+    private void ensureNotificationBelongsToUser(Long notificationId, Long userId) {
+        Notification n = notificationService.findById(notificationId);
+        if (!n.getReceiver().getId().equals(userId)) {
+            throw new AccessDeniedException("Access denied to this notification.");
```
there's a class, `ErrorConstants` or smth like that, which contains messages for exceptions. would be nice to move this one there.

natri0 reviewed (changes requested):
just a few minor nitpicks. otherwise great! please translate the comments to english and we can merge.

Shcherba04 commented on service/src/main/java/greencity/controller/NotificationController.java:
```diff
@@ -0,0 +1,69 @@
+        return ResponseEntity.noContent().build();
+    }
+
+    /** всі нотифікації користувача як прочитані. */
```
```suggestion
    /** All user notification as read */
```

Shcherba04 commented on service/src/main/java/greencity/controller/NotificationController.java:
```diff
@@ -0,0 +1,69 @@
+        return ResponseEntity.noContent().build();
+    }
+
+    /** список нотифікацій як прочитані. */
```
```suggestion
    /** List of notifications as read */
```

oleksandrYeriemieiev committed:
fix: apply reviewer feedback
- EN Javadoc instead of UA
- use ErrorMessage.ACCESS_DENIED_NOTIFICATION
- minor cleanup
